scripts/tests

```diff
@@ -279,21 +300,24 @@ def kat(force_qemu, verbose):
     default="NO",
     help="Method for counting clock cycles. PMU requires (user-space) access to the Arm Performance Monitor Unit (PMU). PERF requires a kernel with perf support.",
 )
-def bench(force_qemu, verbose, cycles):
+@click.option(
+    "--cpu",
+    nargs=1,
+    help="Benchmark on specified CPU",
+)
+def bench(force_qemu, verbose, cycles, cpu):
     config_logger(verbose)
 
     results = test_schemes(
         "benchmark",
         lambda scheme: scheme.name.replace("MLKEM", "test/bin/bench_kyber"),
         lambda _: True,
         lambda _: True,
+        process_bench,
         force_qemu,
         verbose,
-        cycles=cycles,
+        [f"CYCLES={cycles}", f"CFLAGS=-mcpu={cpu}"],
```

Comment:
Should `-mcpu` not be omitted if the `--cpu` argument isn't given?

Reply:
Oh yes! I was trying if this work, and forgot to handle it properly. Thanks for the catch

Reply:
We might also want to omit `-march` when `-mcpu` is specified, as otherwise there may be conflicts. I was just trying this with `-mcpu=neoverse-n1` and got
```
cc1: error: switch ‘-mcpu=neoverse-n1’ conflicts with ‘-march=armv8-a’ switch [-Werror]
```

Reply:
@hanno-becker Where is the `-march` coming from ? We currently don't have this flag set in our `Makefile` right ?

Reply:
btw, on which platform you got this error ? 
I didn't got any error for setting `-mcpu=neoverse-n1` when compiling on my mac

Reply:
I ran it on a Neoverse N1

Reply:
> Where is the -march coming from ? We currently don't have this flag set in our Makefile right ?

Good question, I just assumed we'd set this somewhere in the `Makefile`. Let me check.

Reply:
This is a bit confusing `-march` is indeed not specified anywhere:
```
gcc -mcpu=neoverse-n1 -Wall -Wextra -Wpedantic -Werror -Wmissing-prototypes -Wredundant-decls -Wshadow -Wpointer-arith -Wno-unknown-pragmas -O3 -fomit-frame-pointer -pedantic -I mlkem -I fips202 -I randombytes -DKYBER_K=2 mlkem/kem.c mlkem/indcpa.c mlkem/polyvec.c mlkem/poly.c mlkem/ntt.c mlkem/cbd.c mlkem/reduce.c mlkem/verify.c fips202/keccakf1600.c fips202/fips202.c mlkem/symmetric-shake.c randombytes/randombytes.c test/hal.c test/bench_kyber.c -o test/bin/bench_kyber512
cc1: error: switch ‘-mcpu=neoverse-n1’ conflicts with ‘-march=armv8-a’ switch [-Werror]
cc1: all warnings being treated as errors
```

Interestingly, I only get this in the nix shell, not with the native gcc.

Reply:
What is the gcc version of both environment ?

Reply:
Ok, I also got the same issue as you on the rpi4 (same machine as the selfhost runner)

The `gcc` version in `nix` shell is
```
gcc (GCC) 13.2.0
Copyright (C) 2023 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE
```

While the native `gcc` is
```
gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

Reply:
I guess there is no way around also passing `-march=armv8.2-a` in this case. 
Maybe it's better to just let the user pass in ARCH_FLAGS.

Reply:
@mkannwischer I find this very odd -- if you don't specify `-march` on the command line, `gcc` shouldn't complain about a conflict. Unless in the Nix shell it's somehow defined via an alias..

Reply:
For the moment, we can either (a) replace `-mcpu` by `-mtune`, which does not interfere with `-march`. IIRC, `-mcpu` is a combination of `-march` (derived from the CPU) and `-mtune`. (b) Add an explicit `--arch` parameter similar to `tests`. (c) Allow the user to specify `ARCH_FLAGS`.

I can live with all of them, but prefer (b) or (c).

Reply:
I don't think it's nix shell causing the problem. I've tried to use the same gcc version as nix shell in the native shell environment, and it result in the same error. And I've checked the env vars of nix shell, there is no CFLAGS and ARCH_FLAGS preexists.

I vote for (c), and I pushed a version where each command of `tests` script can set both extra cflags and arch flags.

Reply:
Great, thank you @potsrevennil